internal/io/http/client.go

```diff
@@ -350,20 +351,22 @@ func (cc *ClientConf) parseResponse(ctx api.StreamContext, resp *http.Response,
 		}
 		return nil, nil, nil
 	case "body":
-		payload, err := decode(ctx, c)
+		payloads, err := decode(ctx, c)
 		if err != nil {
 			return nil, c, err
 		}
-		ro := &bodyResp{}
-		err = cast.MapToStruct(payload, ro)
-		if err != nil {
-			return nil, c, fmt.Errorf("invalid body response: %v", err)
-		}
-		if resp.StatusCode < 200 || resp.StatusCode > 299 {
-			return nil, c, fmt.Errorf("http status code is not 200: %v", payload)
+		for _, payload := range payloads {
+			ro := &bodyResp{}
+			err = cast.MapToStruct(payload, ro)
+			if err != nil {
```

Comment:
Not sure if we should fail whole response for one parsing error or still return partial result.

Reply:
I think fail is better, otherwise it may look like we discard some requests.